### PR TITLE
Allow trace header propagation without trace exporting

### PR DESCRIPTION
**Pull request by berndverst**
This change allows trace ID / traceparent to be propagated or generated by the runtime without exporting full trace logs.

In configuration this can be done via

```
apiVersion: dapr.io/v1alpha1
kind: Configuration
metadata:
  name: daprConfig
spec:
  tracing:
    samplingRate: "1"
```

Previously it was necessary to choose one of the existing exporters like `otel`, `zipkin` or `stdout` (and configure an endpoint for the former two) even if all one was interested in was the forwarding of trace headers through Dapr.

Internally this is accomplished by creating a `NullExporter` which is registered if none of the other trace exporters have been registered.

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -408,6 +408,10 @@ func (a *DaprRuntime) setupTracing(hostAddress string, tpStore tracerProviderSto
 		tpStore.RegisterExporter(otelExporter)
 	}
 
+	if !tpStore.HasExporter() && tracingSpec.SamplingRate != "" {
```
Comment (berndverst) [inline pkg/runtime/runtime.go:411]: Note that this entire code (and the code above) won't be used if Sampling Rate is 0 anyway

Comment (ItalyPaleAle) [inline pkg/runtime/runtime.go:411]: Shouldn't this check if the value of SamplingRate is also != "0" ?

Comment (ItalyPaleAle) [reply pkg/runtime/runtime.go:411]: Oh, I saw your comment posted a the same time as mine :)

**Review by ItalyPaleAle — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]

**Pull request merged**